clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java

```diff
@@ -288,9 +331,21 @@ public ConnectionSource getConnectionSource() {
    * Returns {@code true} is server side components should be automatically created.
    *
    * @return {@code true} is auto-create is enabled
+   * @deprecated Deprecated in favor of {@link #getClientMode()}
    */
+  @Deprecated
```

Comment:
why we have to deprecate ? Isn't the method stil doing the exact same thing the javadoc says ?

Reply:
It's return type can't discriminate between the full set of behaviors. Anyone using this should probably be migrating to `getClientMode()` since they may care about the distinction between `AUTO_CREATE` and `AUTO_CREATE_ON_RECONNECT`.

Reply:
but this method meaning is not to get the configuration state but much more to know whether we have to auto-create or not ? It could have been called: `boolean shouldAutoCreate()` right ?

Reply:
This method still isn't good enough in that situation since whether we should 'auto-create' in a given situation is now a function of two things as follows:

mode                                    | connect | reconnect |
---------------------------|---------|------------|
auto-create                          |    true    |      false       |
auto-create-on-reconnect |    true    |       true       |

A single boolean return cannot capture this behavior. As an extension I could clarify the javadoc here to clarify that a true return here means only: "this configuration will auto-create on connect (but might also auto-create at other times).